test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs

```diff
@@ -42,4 +43,90 @@ public void ProcessMetricsAreCaptured()
         var virtualMemoryMetric = exportedItems.FirstOrDefault(i => i.Name == "process.memory.virtual");
         Assert.NotNull(virtualMemoryMetric);
     }
+
+    [Fact]
+    public void CheckValidGaugeValueWhen2MeterProviderInstancesHaveTheSameMeterName()
+    {
+        var exportedItemsA = new List<Metric>();
+        var exportedItemsB = new List<Metric>();
+
+        using var meterProviderA = Sdk.CreateMeterProviderBuilder()
+            .AddProcessInstrumentation()
+            .AddInMemoryExporter(exportedItemsA)
+            .Build();
+
+        using var meterProviderB = Sdk.CreateMeterProviderBuilder()
+            .AddProcessInstrumentation()
+            .AddInMemoryExporter(exportedItemsB)
+            .Build();
+
+        meterProviderA.ForceFlush(MaxTimeToAllowForFlush);
+        meterProviderB.ForceFlush(MaxTimeToAllowForFlush);
+
+        var metricA = exportedItemsA.FirstOrDefault(i => i.Name == "process.memory.usage");
+        var metricB = exportedItemsB.FirstOrDefault(i => i.Name == "process.memory.usage");
+
+        Assert.True(GetValue(metricA) > 0);
+        Assert.True(GetValue(metricB) > 0);
+    }
+
+    [Fact]
+    public void UnifiedMetricStreamIdentityWhen2MeterProviderInstancesHaveTheSameMeterName()
+    {
+        var exportedItemsA = new List<Metric>();
+        var exportedItemsB = new List<Metric>();
+
+        Meter m1 = new("myMeter");
+        Meter m2 = new("myMeter");
+
+        m1.CreateObservableCounter(
+            "myGaugeName",
+            () => { return 1D; },
+            unit: "1",
+            description: "test");
+
+        m1.CreateObservableCounter(
+            "myGaugeName",
+            () => { return 2D; },
+            unit: "1",
+            description: "test");
+
+        using var meterProviderA = Sdk.CreateMeterProviderBuilder()
+            .AddMeter("myMeter")
+            .AddInMemoryExporter(exportedItemsA)
+            .Build();
+
+        using var meterProviderB = Sdk.CreateMeterProviderBuilder()
+            .AddMeter("myMeter")
+            .AddInMemoryExporter(exportedItemsB)
+            .Build();
+
+        meterProviderA.ForceFlush();
+        meterProviderB.ForceFlush();
+
+        var metricA = exportedItemsA.FirstOrDefault(i => i.Name == "myGaugeName");
+        var metricB = exportedItemsB.FirstOrDefault(i => i.Name == "myGaugeName");
+
+        Assert.Equal(GetValue(metricA), GetValue(metricB));
```

Comment:
Do we expect them to be equal?

Reply:
yes, for counters as the MetricStreamIdentity will be the same if the meter name matches, the value will be equal.


While, for gauges, my understanding is that since they do a spontaneous measurement at the time they were being collected, so it is hard to use InMemoryExporter to create a unit test for that.

But I checked that the running values in the aggregationStore match:
![image](https://user-images.githubusercontent.com/16979322/194938230-87141e40-8a19-4ed6-a49b-b16c7f0f87b4.png)


```
((OpenTelemetry.Metrics.Metric)(new System.Linq.SystemCore_EnumerableDebugView<System.Diagnostics.Metrics.ListenerSubscription>((new System.Collections.Generic.ICollectionDebugView<System.Diagnostics.Metrics.Instrument>(((OpenTelemetry.Instrumentation.Process.ProcessMetrics)(new System.Collections.Generic.ICollectionDebugView<object>(((OpenTelemetry.Metrics.MeterProviderSdk)meterProviderA).Instrumentations).Items[0])).MeterInstance._instruments).Items[0])._subscriptions).Items[0]).State).aggStore.metricPoints[0].runningValue.AsDouble



((OpenTelemetry.Metrics.Metric)(new System.Linq.SystemCore_EnumerableDebugView<System.Diagnostics.Metrics.ListenerSubscription>((new System.Collections.Generic.ICollectionDebugView<System.Diagnostics.Metrics.Instrument>(((OpenTelemetry.Instrumentation.Process.ProcessMetrics)(new System.Collections.Generic.ICollectionDebugView<object>(((OpenTelemetry.Metrics.MeterProviderSdk)meterProviderB).Instrumentations).Items[0])).MeterInstance._instruments).Items[0])._subscriptions).Items[0]).State).aggStore.metricPoints[0].runningValue.AsDouble
```

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/664/files#r991558434

You have two ObservableCounters here: one always returns `1` and the other always returns `2` yet this line: `Assert.Equal(GetValue(metricA), GetValue(metricB));` succeeds. This means that only one of the ObservableCounters callbacks is being called.

Reply:
Hmm... what I see is that both callbacks will be triggered for the meterProviders, but the second callback will hit this:
![image](https://user-images.githubusercontent.com/16979322/194942118-fbb8bb51-81d5-4f6a-a73e-e641213fc5e7.png)

test code here:
https://github.com/open-telemetry/opentelemetry-dotnet/compare/main...Yun-Ting:yunl/instrumentNames?expand=1